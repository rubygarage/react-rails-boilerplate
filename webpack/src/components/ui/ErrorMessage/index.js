import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Field } from 'redux-form';

class ErrorMessage extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
  }

  field = ({ meta: { touched, error } }) => (
    <div className={classNames({ hidden: !(touched && error) }, 'alert', 'alert-danger')}>
      { touched && error && <p className="error-message">{error}</p> }
    </div>
  )

  render() {
    const { name } = this.props;

    return (
      <Field name={name} component={this.field} />
    );
  }
}

export default ErrorMessage;
