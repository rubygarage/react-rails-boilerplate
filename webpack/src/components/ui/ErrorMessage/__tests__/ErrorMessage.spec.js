import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from 'components/ui/ErrorMessage';

describe('<ErrorMessage />', () => {
  const props = {
    name: 'TestFieldName',
    component: jest.fn(),
  };

  it('renders with empty props', () => {
    const component = shallow(<ErrorMessage />);

    expect(component).toMatchSnapshot();
  });

  it('renders with props', () => {
    const component = shallow(<ErrorMessage {...props} />);

    expect(component).toMatchSnapshot();
  });
});
