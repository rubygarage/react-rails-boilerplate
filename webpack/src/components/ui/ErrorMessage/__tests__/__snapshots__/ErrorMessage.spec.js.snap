// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> renders with empty props 1`] = `
<Field
  component={[Function]}
  name=""
/>
`;

exports[`<ErrorMessage /> renders with props 1`] = `
<Field
  component={[Function]}
  name="TestFieldName"
/>
`;
